ra/ra.go

```diff
@@ -1343,3 +1349,48 @@ func (ra *RegistrationAuthorityImpl) DeactivateAuthorization(ctx context.Context
 	}
 	return nil
 }
+
+// NewOrder creates a new order object
+func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.NewOrderRequest) (*corepb.Order, error) {
+	expires := ra.clk.Now().Add(ra.orderLifetime).UnixNano()
+	status := string(core.StatusPending)
+	order := &corepb.Order{
+		RegistrationID: req.RegistrationID,
+		Expires:        &expires,
+		Csr:            req.Csr,
+		Status:         &status,
+	}
+	parsedCSR, err := x509.ParseCertificateRequest(req.Csr)
+	if err != nil {
+		return nil, err
+	}
+
+	err = csr.VerifyCSR(parsedCSR, ra.maxNames, &ra.keyPolicy, ra.PA, ra.forceCNFromSAN, *req.RegistrationID)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, name := range parsedCSR.DNSNames {
+		authz, err := ra.NewAuthorization(ctx, core.Authorization{
```

Comment:
This will generate N roundtrips to the SA, one for each authorization. Proposal: Add a AddOrGetAuthorizations batch method to SA, and batch these up as a single request. That also prepares us for future implementation of authz recycling in the new-order flow (please file a ticket for that).